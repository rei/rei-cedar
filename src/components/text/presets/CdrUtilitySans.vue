<script setup lang="ts">
import { useCssModule, computed, type PropType } from 'vue';
import CdrText from '../CdrText.vue';

defineOptions({
  name: 'CdrUtilitySans',
});


type scaleValue = '-1'|'0'|'1'|'2'|'3';
const props = defineProps({
  /** 
   * Sets the type scale
   * @type scaleValue
   * @values '-1','0','1','2','3'
   */ 
  scale: { type: String as PropType<scaleValue>, default: '1' },
  /** Toggles the strong variant */
  strong: { type: Boolean, default: false }
});

const typeProperties = computed(() => {
  return {
    '--cdr-utility-sans-font-size': `var(--cdr-type-scale-${props.scale})`,
    '--cdr-utility-sans-line-height': 
      `calc(var(--cdr-type-scale-${props.scale}) * var(--cdr-utility-sans-line-height-ratio))`,
    '--cdr-utility-sans-font-weight': props.strong ? '600' : '400',

  }
});

const baseClass = 'cdr-utility-sans';
const style = useCssModule();
</script>

<template>
  <CdrText
    :class="style[baseClass]"
    :style="typeProperties"
  >
    <slot />
  </CdrText>


</template>

<style module lang="scss">
@import "./styles/CdrPresets.module.scss";
@import "../../../styles/fluid.css";

.cdr-utility-sans {
  @include cdr-text-base-mixin;
  @include cdr-text-utility-sans-800;
  font-size: var(--cdr-utility-sans-font-size);
  line-height: var(--cdr-utility-sans-line-height);
  font-weight: var(--cdr-utility-sans-font-weight);
}
</style>  