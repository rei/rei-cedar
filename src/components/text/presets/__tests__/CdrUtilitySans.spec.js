import { mount } from '../../../../../test/vue-jest-style-workaround.js';
import CdrUtilitySans from '../CdrUtilitySans.vue';


describe('CdrUtilitySans', () => {
  describe('default configuration', () => {
    let wrapper;
    beforeEach(() => {
      wrapper = mount(CdrUtilitySans, {
        slots: {
          default: 'paragraph'
        }
      });
    });
    it('renders correctly', () => {
      expect(wrapper.element).toMatchSnapshot();
    });
  });

  describe('passing a scale', () => {
    let wrapper;
    beforeEach(() => {
      wrapper = mount(CdrUtilitySans, {
        props: { scale: '-1' },
        slots: {
          default: 'paragraph'
        }
      });
    });
    it('renders correctly', () => {
      expect(wrapper.element).toMatchSnapshot();
    });
  });

  describe('setting to strong variant', () => {
    let wrapper;
    beforeEach(() => {
      wrapper = mount(CdrUtilitySans, {
        props: { strong: true },
        slots: {
          default: 'paragraph'
        }
      });
    });
    it('renders correctly', () => {
      expect(wrapper.element).toMatchSnapshot();
    });
  });

});
