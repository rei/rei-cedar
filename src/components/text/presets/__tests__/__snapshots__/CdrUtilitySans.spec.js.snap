// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CdrUtilitySans > default configuration > renders correctly 1`] = `
<p
  class="cdr-utility-sans"
  style="--cdr-utility-sans-font-size: var(--cdr-type-scale-1); --cdr-utility-sans-line-height: var(--cdr-line-height-ratio-utility-1); --cdr-utility-sans-font-weight: 400;"
>
  
  paragraph
  
</p>
`;

exports[`CdrUtilitySans > passing a scale > renders correctly 1`] = `
<p
  class="cdr-utility-sans"
  style="--cdr-utility-sans-font-size: var(--cdr-type-scale--1); --cdr-utility-sans-line-height: var(--cdr-line-height-ratio-utility--1); --cdr-utility-sans-font-weight: 400;"
>
  
  paragraph
  
</p>
`;

exports[`CdrUtilitySans > setting to strong variant > renders correctly 1`] = `
<p
  class="cdr-utility-sans"
  style="--cdr-utility-sans-font-size: var(--cdr-type-scale-1); --cdr-utility-sans-line-height: var(--cdr-line-height-ratio-utility-1); --cdr-utility-sans-font-weight: 600;"
>
  
  paragraph
  
</p>
`;
